themes/default/content/blog/kubernetes-fundamentals-part-two/index.md

```diff
@@ -0,0 +1,215 @@
+```
+
+## Creating a Cluster with Google Kubernetes Engine
+
+Next, we need a cluster. Start by creating a new directory, navigating into it, and initializing a new Pulumi project using typescript.
+
+```bash
+mkdir gke-hello-world && cd gke-hello-world && pulumi new typescript
+```
+
+We also need to install some dependencies, and set some required configuration values.
+
+```bash
+npm install --save @pulumi/pulumi @pulumi/gcp @pulumi/kubernetes
+pulumi config set gcp:project <YOUR_GCP_PROJECT_HERE>
+pulumi config set gcp:zone us-west1-a     // any valid GCP Zone here
+```
+
+Now, in `index.ts` add the following code:
+
+```typescript
+import * as k8s from "@pulumi/kubernetes";
+import * as pulumi from "@pulumi/pulumi";
+import * as gcp from "@pulumi/gcp";
+
+const name = "helloworld";
+
+// Create a GKE cluster
+const engineVersion = gcp.container.getEngineVersions().then(v => v.latestMasterVersion);
+const cluster = new gcp.container.Cluster(name, {
+    initialNodeCount: 2,
+    minMasterVersion: engineVersion,
+    nodeVersion: engineVersion,
+    nodeConfig: {
+        machineType: "n1-standard-1",
+        oauthScopes: [
+            "https://www.googleapis.com/auth/compute",
+            "https://www.googleapis.com/auth/devstorage.read_only",
+            "https://www.googleapis.com/auth/logging.write",
+            "https://www.googleapis.com/auth/monitoring"
+        ],
+    },
+});
+
+// Export the Cluster name
+export const clusterName = cluster.name;
+
+// Manufacture a GKE-style kubeconfig. Note that this is slightly "different"
+// because of the way GKE requires gcloud to be in the picture for cluster
+// authentication (rather than using the client cert/key directly).
+export const kubeconfig = pulumi.
+    all([ cluster.name, cluster.endpoint, cluster.masterAuth ]).
+    apply(([ name, endpoint, masterAuth ]) => {
+        const context = `${gcp.config.project}_${gcp.config.zone}_${name}`;
+        return `apiVersion: v1
+clusters:
+- cluster:
+    certificate-authority-data: ${masterAuth.clusterCaCertificate}
+    server: https://${endpoint}
+  name: ${context}
+contexts:
+- context:
+    cluster: ${context}
+    user: ${context}
+  name: ${context}
+current-context: ${context}
+kind: Config
+preferences: {}
+users:
+- name: ${context}
+  user:
+    auth-provider:
+      config:
+        cmd-args: config config-helper --format=json
+        cmd-path: gcloud
+        expiry-key: '{.credential.token_expiry}'
+        token-key: '{.credential.access_token}'
+      name: gcp
+`;
+    });
+
+// Create a Kubernetes provider instance that uses our cluster from above.
+const clusterProvider = new k8s.Provider(name, {
+    kubeconfig: kubeconfig,
+});
+```
+
+This is the entrypoint for our program, and is what's setting up the cluster for us. We're defining the resources we want, the `kubeconfig` that'll be used to access the cluster, and also the Kubernetes provider so that we can deploy resources to it next.
```

Comment:
```suggestion
This code is the entrypoint for our program and is what's setting up the cluster for us. We're defining the resources we want, the `kubeconfig` that'll be used to access the cluster, and also the Kubernetes provider so that we can deploy resources to it next.
```